src/zenml/pipelines/base_pipeline.py

```diff
@@ -260,7 +271,40 @@ def requirements(self) -> Set[str]:
                     f"'{integration_name}'."
                 ) from e
 
-        if self.requirements_file and fileio.exists(self.requirements_file):
+        if (
+            isinstance(self._requirements, str)
+            and self._requirements
+            and fileio.exists(self._requirements)
+        ):
+            with fileio.open(self._requirements, "r") as f:
+                requirements.update(
+                    {
+                        requirement.strip()
+                        for requirement in f.read().split("\n")
+                    }
+                )
+        elif isinstance(self._requirements, str) and self._requirements:
+            root = str(Repository().root)
+            if root:
+                assumed_requirements_file = os.path.join(
+                    root, "requirements.txt"
+                )
+                if fileio.exists(assumed_requirements_file):
+                    with fileio.open(assumed_requirements_file, "r") as f:
+                        requirements.update(
+                            {
+                                requirement.strip()
+                                for requirement in f.read().split("\n")
+                            }
+                        )
+                        logger.info(
+                            "Using requirements file: `%s`",
+                            assumed_requirements_file,
+                        )
+        elif isinstance(self._requirements, List) and self._requirements:
```

Comment:
Can remove the `self._requirements` here as well

Reply:
I guess if it's an empty list it doesn't matter if it enters this conditional branch?

Reply:
Yep I think so, will just not add anything to the requirements